src/NuGet.Core/NuGet.Commands/RestoreCommand/SourceRepositoryDependencyProvider.cs

```diff
@@ -239,9 +239,17 @@ public class SourceRepositoryDependencyProvider : IRemoteDependencyProvider
                     };
                 }
             }
-            catch(FatalProtocolException e) when(_ignoreFailedSources)
+            catch (FatalProtocolException e)
             {
-                await LogWarningAsync(libraryRange.Name, e);
+                if (_ignoreFailedSources)
+                {
+                    await LogWarningAsync(libraryRange.Name, e);
+                }
+                else
+                {
+                    RestoreLogMessage logMessage = RestoreLogMessage.CreateError(e.LogCode, ExceptionUtilities.DisplayMessage(e));
+                    await logger.LogAsync(logMessage);
```

Comment:
yes, expanding my test case. NU1301,3,4,7 are raised 1 time per top level package reference.
how do we generally do consolidation of errors like this?
will investigate how to decrease.

Reply:
working on fix. looks promising.

Reply:
So I was thinking about this this morning, and it might be desirable that you throw here and stop the operation. 

If we don't stop the operation when a package source fails, we are effectively ignoring failed sources. 

Basically, it's possible that we download a bunch of packages the first time (from other sources) and then later when you restore again, we use those packages from the global packages folder/http cache and restore succeeds (sort of, there's a chance it's not repeatable).